### PR TITLE
Introduce a safe mechanism for adding arbitrary text to an IRC command/&Line{}

### PR DESCRIPTION
**Pull request by liamzdenek**
#6

Considerations:
- Preserve reverse-compatibility. I ultimately did not do this, as NewLine() was renamed to NewLineFromRaw(), however, older code should be able to simply use this function in place, and everything should continue to work. This is an acceptable compromise in exchange for clarity, IMO
- Preserve an escape hatch for arbitrary code. Sometimes, validation is a waste, and I would like to opt out sometimes. In its current form, that escape hatch is via a &Line{} literal.
- Selection of the correct pattern. I chose the builder pattern (even though it is a bit wordy for such a simple use case) in exchange for modularity, and brevity of end-user code.
- re prev: Optional sanitization. I would like the end user to be able to explicitly .Sanitize() to clean it up, or neglect it and catch an error instead.

### CODE REVIEW

**Comment by tsdko** [comment]
I'm not sure what advantage there is in using a Builder pattern in Go instead of just having a constructor for `Line` and a `Sanitize()` method, otherwise LGTM.

**Comment by liamzdenek** [comment]
Opting out of certain arguments, eg, prefix:
`NewLine("", "PRIVMSG", []string{"Liam"}, "Hello");` encodes to `PRIVMSG Liam :Hello`
`NewLine("", "EOS", []string{}, "")` encodes to `EOS`
`NewLineBuilder().Command("PRIVMSG").PushArg("Liam").Suffix("Hello").Consume()`
`NewLineBuilder().Command("EOS").Consume()`

I always have to provide a prefix, suffix, and argument list even if I do not require one. I also have to remember the order of the arguments and which they mean. I would rather type more in exchange for clarity, and ensure that everything i type has meaning, eg, `""` for an unused field is something I would like to avoid.
